src/app/qgsmaptoolselectutils.cpp

```diff
@@ -335,3 +337,193 @@ QgsFeatureIds QgsMapToolSelectUtils::getMatchingFeatures( QgsMapCanvas *canvas,
   return newSelectedFeatures;
 }
 
+
+QgsMapToolSelectUtils::QgsMapToolSelectMenuActions::QgsMapToolSelectMenuActions(
+  QgsMapCanvas *canvas,
+  QgsVectorLayer *vectorLayer,
+  QgsVectorLayer::SelectBehavior behavior,
+  QObject *parent ):
+  QObject( parent ),
+  mCanvas( canvas ),
+  mVectorLayer( vectorLayer ),
+  mBehavior( behavior )
+{
+  connect( mVectorLayer, &QgsMapLayer::destroyed, this, &QgsMapToolSelectMenuActions::onLayerDestroyed );
+}
+
+QList<QAction *> QgsMapToolSelectUtils::QgsMapToolSelectMenuActions::actions( const QgsFeatureIds &featureCanditateIds )
+{
+  QList<QAction *> actionsList;
+
+  QString beginningOfText;
+
+  QgsFeatureIds effectiveFeatureIds = featureCanditateIds;
+
+  switch ( mBehavior )
+  {
+    case QgsVectorLayer::SetSelection:
+      beginningOfText = tr( "Select " );
+      break;
+    case QgsVectorLayer::AddToSelection:
+    {
+      beginningOfText = tr( "Add " );
```

Comment:
This is hard to interpret without context, what do you think about "Add %1 to selection", "Remove %1 from selection"?